packages/atomic/src/utils/accessibility-utils.tsx

```diff
@@ -17,7 +18,58 @@ export function AriaLiveRegion(regionName: string) {
 
   return (component: InitializableComponent, setterName: string) => {
     Object.defineProperty(component, setterName, {
-      set: (message) => dispatchMessage(message),
+      set: (message: string) => dispatchMessage(message),
     });
   };
 }
+
+export interface FocusTargetController {
+  setTarget(element: HTMLElement | undefined): void;
+  focusAfterSearch(): void;
+}
+
+export function FocusTarget() {
+  return (component: InitializableComponent, setterName: string) => {
+    const {componentWillLoad} = component;
+
+    component.componentWillLoad = function () {
+      componentWillLoad && componentWillLoad.call(this);
+      const {componentDidRender} = this;
+      let focusAfterSearch = false;
+      let lastSearchId: string | undefined = undefined;
+      let element: HTMLElement | undefined = undefined;
+
+      function tryFocusElement(bindings: Bindings) {
+        if (!element) {
+          return;
+        }
+        const searchId = bindings.engine.state.search.searchResponseId;
+        if (focusAfterSearch && searchId !== lastSearchId) {
+          setTimeout(() => element?.focus());
```

Comment:
This `setTimeout` is unfortunately necessary. I believe it's because some elements aren't appended to the DOM yet.